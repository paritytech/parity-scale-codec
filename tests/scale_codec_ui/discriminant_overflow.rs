#[derive(::parity_scale_codec::Decode, ::parity_scale_codec::Encode)]
#[codec(crate = ::parity_scale_codec)]
enum T {
	A = 256,
}

fn main() {}
