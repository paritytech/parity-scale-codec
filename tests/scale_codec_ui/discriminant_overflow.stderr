error[E0080]: evaluation of constant value failed
 --> tests/scale_codec_ui/discriminant_overflow.rs:1:10
  |
1 | #[derive(::parity_scale_codec::Decode, ::parity_scale_codec::Encode)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'Discriminant must be in the range 0..255, variant `A` index is `256`.', $DIR/tests/scale_codec_ui/discriminant_overflow.rs:1:10
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `::core::panic` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of constant value failed
 --> tests/scale_codec_ui/discriminant_overflow.rs:1:40
  |
1 | #[derive(::parity_scale_codec::Decode, ::parity_scale_codec::Encode)]
  |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'Discriminant must be in the range 0..255, variant `A` index is `256`.', $DIR/tests/scale_codec_ui/discriminant_overflow.rs:1:40
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `::core::panic` (in Nightly builds, run with -Z macro-backtrace for more info)
